Port pipeline to DSL2

christopher-mohr opened the pull request:
This
- ports the `hlatyping` pipeline to DSL2
- uses existing nf-core modules
- merges nf-core template updates until `2.6`
- switches to samplesheet input 
- adds functionality for using `fastq` and `bam` files in the same run.

We still need a better solution for handling the HLA reference. Currently it is only possible to either process `dna` or `rna` data at the same time due to the HLA reference that has to be provided. My suggestion would be to choose the `.fasta` automatically based on the specified `seq_type` in the sample sheet AND allow users to specify custom `.fasta` files in the sample sheet (this might need then an additional process because of the `alleles.h5` file). 

In order to allow the use of different HLA references in the same run we also need to update the `yara_index` module and at least add the `meta` map to the process I guess. @apeltzer what do you think?

<!--
# nf-core/hlatyping pull request

Many thanks for contributing to nf-core/hlatyping!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/hlatyping/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated
- [x] If necessary, also make a PR on the [nf-core/hlatyping branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/hlatyping)

apeltzer commented:
Nice :) 
Yes, please work on an updated yara module - we've worked on that one when modules were started so some stuff wasn't fully clear back then on how to work with it. So an update of the module would be necessary I fear. 

Once you updated the module, I can review this here in detail

apeltzer commented:
Please ping me on the yara module then @christopher-mohr

christopher-mohr commented:
I updated the `yara` modules to the latest version and removed the `FASTA` parameter. The HLA references are now automatically set based on the given sequencing type. Not sure if there is an easier/ more elegant way for making sure that the reads are matched with the right index.

In addition I would add the possibility to provide a custom `FASTA` (for a specific sample) in the sample sheet in a separate PR.

apeltzer commented on Dockerfile:
```diff
@@ -1,6 +1,6 @@
 FROM nfcore/base:1.10.2
 LABEL authors="Christopher Mohr, Alexander Peltzer, Sven Fillinger" \
-      description="Docker image containing all software requirements for the nf-core/hlatyping pipeline"
```
We should ditch that entirely - all modules have their own now, so can just remove the dockerfile entirely.

apeltzer commented on README.md:
```diff
@@ -1,75 +1,87 @@
+The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The [Nextflow DSL2](https://www.nextflow.io/docs/latest/dsl2.html) implementation of this pipeline uses one container per process which mak ...
 
-## Quick Start
+<!-- TODO nf-core: Add full-sized test dataset and amend the paragraph below if applicable -->
```
Remove please

apeltzer commented on README.md:
```diff
@@ -1,75 +1,87 @@
+## Credits
 
-![DAG with `.bam` file](docs/images/hlatyping_dag_bam.png)
+nf-core/hlatyping was originally written by [Christopher Mohr](https://github.com/christopher-mohr) from [Medical Data Integration Center](https://www.medizin.uni-tuebingen.de/de/das-klinikum/einrichtungen/institute/informationstechnologie-und-medizininformatik/medic) and [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qb ...
```
```suggestion
nf-core/hlatyping was originally written by [Christopher Mohr](https://github.com/christopher-mohr) from [Medical Data Integration Center](https://www.medizin.uni-tuebingen.de/de/das-klinikum/einrichtungen/institute/informationstechnologie-und-medizininformatik/medic) and [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/), [Alexander Peltzer](https://github.com/apeltzer) from [Boehringer Ingelheim](https://www.boehringer-ingelheim.de), and [Sven Fillinger](https://github.com/sven1103) from [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/).
```

apeltzer commented on README.md:
```diff
@@ -1,75 +1,87 @@
+We thank the following people for their extensive assistance in the development of this pipeline:
 
-nf-core/hlatyping was originally written by [Christopher Mohr](https://github.com/christopher-mohr) from [Institute for Translational Bioinformatics](https://kohlbacherlab.org/team_tbi/) and [Quantitative Biology Center](https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/),  [Alexander Peltzer](https://github.com/apeltzer) from [Böhringer Ingelheim] ...
+<!-- TODO nf-core: If applicable, make list of people who have also contributed -->
```
```suggestion
```

Or do we have other contributors who we should list here?

christopher-mohr replied:
Let's leave it in there for now, so we don't forget it.

apeltzer commented on README.md:
```diff
@@ -1,75 +1,87 @@
-## Citation
+## Citations
+
+<!-- TODO nf-core: Add citation for pipeline after first release. Uncomment lines below and update Zenodo doi and badge at the top of this file. -->
```
```suggestion
```

apeltzer commented on assets/methods_description_template.yml:
```diff
@@ -0,0 +1,25 @@
+section_name: "nf-core/hlatyping Methods Description"
+section_href: "https://github.com/nf-core/hlatyping"
+plot_type: "html"
+## TODO nf-core: Update the HTML below to your prefered methods description, e.g. add publication citation for this pipeline
```
```suggestion
```

apeltzer commented on docs/usage.md:
```diff
@@ -1,23 +1,85 @@
+
 ## Introduction
 
+The hlatyping pipeline can currently deal with two input formats: `.fastq{.gz}` or `.bam`, not both at the same time however. If the input file type is `bam`, than the pipeline extracts all reads from it and performs an mapping additional step with the `yara` mapper against the HLA reference sequence. Indices are provided in the `./data` directory of this repository. Optitype uses [razers3](https ...
```
Is this still true? I thought the new version can handle both ...?

apeltzer replied:
Indices are not in .data anymore - please remove that part :-)

apeltzer reviewed (approved):
Looks good to have this on DSL2 now. Some minor comments. I will also run once this is on dev to see if docs are matching expectations 👍🏻
